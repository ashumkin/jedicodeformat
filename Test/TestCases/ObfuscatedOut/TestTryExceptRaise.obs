unit testtryexceptraise;interface implementation uses sysutils;procedure bar;begin
end;procedure signalerror;begin end;procedure foo;begin try bar;except on eabort 
do raise;else signalerror;raise;end;end;procedure fish;begin try bar;except on eabort
do raise;else signalerror;bar;foo;bar;raise;end;end;end.