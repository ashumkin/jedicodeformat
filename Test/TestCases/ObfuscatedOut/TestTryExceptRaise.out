unit testtryexceptraise;

interface

implementation

uses SysUtils;

procedure bar;
begin
end;

procedure signalerror;
begin
end;

procedure foo;
begin
  try
    bar;
  except
    on eabort do
      raise;
    else
      signalerror;
    raise;
  end;
end;

procedure fish;
begin
  try
    bar;
  except
    on eabort do
      raise;
    else
      signalerror;
    bar;
    foo;
    bar;
    raise;
  end;
end;

end.
